### PR TITLE
Refactor hy.models

### PR DESCRIPTION
**Pull request by Kodiologist**

### CODE REVIEW

File: hy/contrib/multi.hy [hunk 1]
```diff
@@ -21,9 +21,7 @@
 ;; DEALINGS IN THE SOFTWARE.
 
 (import [collections [defaultdict]]
-        [hy.models.expression [HyExpression]]
-        [hy.models.list [HyList]]
-        [hy.models.string [HyString]])
+        [hy [HyExpression HyList HyString]])
```
Comment (gilch) [inline hy/contrib/multi.hy]: I'm confused by this. Shouldn't it be `[hy.models [HyExpression HyList HyString]]`? And yet the tests pass. Are we not testing this?

**Comment by gilch** [comment]
The docs would also have to be updated to match the new structure. Also, is this really an improvement? I'm not sure. There was no discussion. Hy really could use some refactoring though.

File: hy/contrib/multi.hy (hunk 1 above)
Comment (kirbyfan64) [inline hy/contrib/multi.hy]: @gilch Hy re-exports them in the main module: https://github.com/hylang/hy/blob/master/hy/__init__.py

**Comment by kirbyfan64** [comment]
IMO it is: the models are rather short, and placing them in separate files really didn't do much except make it a little trickier to maneuver through.

**Comment by Kodiologist** [comment]
> The docs would also have to be updated to match the new structure.

Oh, whoops, I assumed that none of it was documented. I'll do that now.

**Comment by Kodiologist** [comment]
Done.

**Comment by gilch** [comment]
I'll approve it then. 👍 

Is this still small enough to squash? I've seen some disagreement on when that's appropriate.

**Comment by Kodiologist** [comment]
I made it into three separate commits because I think it's nice to have the reorganization separate from the slightly more meaningful changes.

**Pull request merged**